docs/Bridging_Assets.md

```diff
@@ -0,0 +1,158 @@
+
+#### Setting up a custom token pairing
+
+**1. Deploy L2 custom token contract**
+
+Start by deploying the L2 token contract [directly onto Arbitrum](Contract_Deployment.md). A custom L2 contract should conform to the minimal IArbToken interface:
+
+```sol
+/**
+ * @title Minimum expected interface for L2 token that interacts with the L2 token bridge (this is the interface necessary
+ * for a custom token that interacts with the bridge, see TestArbCustomToken.sol for an example implementation).
+ */
+interface IArbToken {
+    /**
+     * @notice should increase token supply by amount, and should (probably) only be callable by the L1 bridge.
+     */
+    function bridgeMint(address account, uint256 amount) external;
+
+    /**
+     * @notice should decrease token supply by amount, and should (probably) only be callable by the L1 bridge.
+     */
+    function bridgeBurn(address account, uint256 amount) external;
+
+    /**
+     * @notice withdraw user tokens from L2 to the L1
+     */
+    function withdraw(address account, uint256 amount) external;
+}
+```
+
+**2. Deploy L1 custom token contract**
+
+The L1 contract should conform to the minimal `ICustomToken` interface:
+
+```sol
+/**
+ * @title Minimum expected interface for L1 custom token (see TestCustomTokenL1.sol for an example implementation)
+ */
+interface ICustomToken {
+    /**
+     * @notice Should make an external call to EthERC20Bridge.registerCustomL2Token
+     */
+    function registerTokenOnL2(
+        address l2CustomTokenAddress,
+        uint256 maxSubmissionCost,
+        uint256 maxGas,
+        uint256 gasPriceBid,
+        address refundAddress
+    ) external virtual;
+
+    function transferFrom(
+        address sender,
+        address recipient,
+        uint256 amount
+    ) external virtual returns (bool);
+
+    function balanceOf(address account) external view virtual returns (uint256);
+}
+```
+
+**3. Register L1/L2 custom pairing**
+
+Calling `MyCustomL1Token.registerTokenOnL2` should make an external call to `EthERC20Bridge.registerCustomL2Token` with the custom L2 token's address (see [TestCustomTokenL1](https://github.com/OffchainLabs/arbitrum/blob/develop/packages/arb-bridge-peripherals/contracts/tokenbridge/test/TestCustomTokenL1.sol) for an example implementation). Once registered, calling `EthERC20Bridge.deposit` with a custom token's L1 address with deposit into the corresponding custom L2 pairing.
+
+Note that [arb-ts](qqq.md) provides client side convenience methods for the functionality listed above, and more.
+
+See [integration tests](https://github.com/OffchainLabs/arbitrum/blob/master/packages/arb-ts/integration_test/arb-bridge.test.ts) for example usage.
+
+## ERC-721 Bridge
+
+TODO
```

Comment:
We could say the ERC20 bridge logic is pretty close to whats needed here, and we're open to community contributions, ie `buidl for us pls, ser`

Reply:
lol let's update it and assess where things are at when the time comes